include/dd/UniqueTable.hpp

```diff
@@ -100,37 +100,19 @@ namespace dd {
                 assert(edge.p->v == v - 1 || edge.isTerminal());
             }
 
-            Node* p = tables[static_cast<std::size_t>(v)][key];
-            while (p != nullptr) {
-                if (nodesAreEqual(e.p, p)) {
-                    // Match found
-                    if (e.p != p && !keepNode) {
-                        // put node pointed to by e.p on available chain
-                        returnNode(e.p);
-                    }
-                    hits++;
-
-                    // variables should stay the same
-                    assert(p->v == e.p->v);
-
-                    // successors of a node shall either have successive variable numbers or be terminals
-                    for ([[maybe_unused]] const auto& edge: e.p->e) {
-                        assert(edge.p->v == v - 1 || edge.isTerminal());
-                    }
-
-                    return {p, e.w};
-                }
-                collisions++;
-                p = p->next;
-            }
+            // Search bucket in table corresponding to hashed value for the given node
+            const auto hashedNode = searchTable(e, key, keepNode);
 
             // node was not found -> add it to front of unique table bucket
-            e.p->next                                = tables[static_cast<std::size_t>(v)][key];
-            tables[static_cast<std::size_t>(v)][key] = e.p;
-            nodeCount++;
-            peakNodeCount = std::max(peakNodeCount, nodeCount);
+            if (hashedNode == Edge<Node>::zero) {
+                e.p->next                                = tables[static_cast<std::size_t>(v)][key];
+                tables[static_cast<std::size_t>(v)][key] = e.p;
+                nodeCount++;
+                peakNodeCount = std::max(peakNodeCount, nodeCount);
 
-            return e;
+                return e;
+            }
+            return hashedNode;
```

Comment:
Tiny nit: I think it's cleaner to return early in case `hashedNode != Edge<Node>::zero`. Saves a bit of indentation and more logically follows from the function call. If something was found -> return it. Otherwise, some more stuff has to be done.

Reply:
Agreed, done. Also add some minor changes to the comments for consistency

Reply:
You can go one step further (as also pointed out by the cpp-linter) and eliminate the `else`.

Reply:
Woops, done